lib/query/query_payments.rb

```diff
@@ -232,6 +232,20 @@ def self.select_to_filter_search(npo_id, query)
 
     end
 
+    if query[:anonymous].present?
+      expr = if(query[:anonymous] == 'true')
+        expr.where(
+          '(coalesce(supporters.anonymous, false) = $anonymous OR coalesce(donations.anonymous, false) = $anonymous)',
+          anonymous: query[:anonymous]
+        )
```

Comment:
We'll need to add an index on anonymous for supporters and donations for this; otherwise it'll be way too slow on large nonprofits.